skyllh/core/minimizer.py

```diff
@@ -117,6 +120,173 @@ def is_repeatable(self, status):
         pass
 
 
+class ScipyMinimizerImpl(MinimizerImpl):
+    """Wrapper for `scipy.optimize.minimize`"""
+
+    def __init__(self, method: str) -> None:
+        super().__init__()
+        self._method = method
+
+    def minimize(self, initials, bounds, func, func_args=None, **kwargs):
+        """Minimizes the given function ``func`` with the given initial function
+        argument values ``initials``.
+
+        Parameters
+        ----------
+        initials : 1D numpy ndarray
+            The ndarray holding the initial values of all the fit parameters.
+        bounds : 2D (N_fitparams,2)-shaped numpy ndarray
+            The ndarray holding the boundary values (vmin, vmax) of the fit
+            parameters.
+        func : callable
+            The function that should get minimized.
+            The call signature must be
+
+                ``__call__(x, *args)``
+
+            The return value of ``func`` must be (f, grads), the function value
+            at the function arguments ``x`` and the ndarray with the values of
+            the function gradient for each fit parameter, if the
+            ``func_provides_grads`` keyword argument option is set to True.
+            If set to False, ``func`` must return only the function value.
+        func_args : sequence | None
+            Optional sequence of arguments for ``func``.
+
+        Additional Keyword Arguments
+        ----------------------------
+        Additional keyword arguments include options for this minimizer
+        implementation. Possible options are:
+
+            func_provides_grads : bool
+                Flag if the function ``func`` also returns its gradients.
+                Default is ``True``.
+
+        Any additional keyword arguments are passed on to the underlaying
+        :func:`scipy.optimize.minimize` minimization function.
+
+        Returns
+        -------
+        xmin : 1D ndarray
+            The array containing the function arguments at the function's
+            minimum.
+        fmin : float
+            The function value at its minimum.
+        status : dict
+            The status dictionary with information about the minimization
+            process. The following information are provided:
+
+            niter : int
```

Comment:
I believe that "niter" is incorrect here. Scipy minimizer usually return "nit". That's why I added the get_niter method to the minimizer implementation interface to abstract the different content of the status dictionary.

Reply:
fixed